src/components/loader-spinner/loader-spinner.stories.mdx

```diff
@@ -0,0 +1,116 @@
+import { Meta, Story, Canvas } from "@storybook/addon-docs";
+import { LoaderSpinner } from "."
+import * as stories from "./loader-spinner.stories.tsx";
+import StyledSystemProps from "../../../.storybook/utils/styled-system-props";
+
+<Meta
+  title="Loader Spinner"
+  parameters={{ info: { disable: true }, controls: { disabled: true } }}
+/>
+
+# Loader Spinner
+
+<a
+  target="_blank"
+  href="https://zeroheight.com/2ccf2b601/p/74be25-loader/b/81df60"
+  style={{ color: '#007E45', fontWeight: 'bold', textDecoration: 'underline' }}
+>
+  Product Design System component
+</a>
+
+Use the loader spinner to help users understand why they are waiting.
+
+Can be used to demonstrate a known or unknown wait time.
+
+## Contents
+
+- [Quick Start](#quick-start)
+- [Examples](#examples)
+- [Props](#props)
+
+## Quick Start
+
+```js
+import LoaderSpinner from "carbon-react/lib/components/loader-spinner";
+```
+
+## Examples
+
+### Default
+
+Please see the default render of the component below. If a user prefers reduced motion, only the
+loading label will be displayed.
+
+<Canvas>
+  <Story name="default" story={stories.DefaultStory}/>
+</Canvas>
+
+### Sizes
+
+Please find an examples of all spinner sizes below. Ranging from sizes `XS` to `XL`.
+
+<Canvas>
+  <Story name="sizes" story={stories.Sizes}/>
+</Canvas>
+
+### Hide Spinner label
+
+When the `showSpinnerLabel` prop is set to `false` no visual label will be rendered underneath the spinner.
+A visually hidden label will still be available for assistive technologies. However, even with `hideSpinnerLabel`
+set to true, a label will still be rendered if end users have prefered reduced motion.
+
+<Canvas>
+  <Story name="hide spinner label" story={stories.HideSpinnerLabel}/>
+</Canvas>
+
+### Colors
+
+Please find an examples of all spinner colours below. Shown against the ligher background are the colors `action` and `neutral`.
+Against the darker background are the `inverse`, `gradient-grey` and `gradient-white` colors.
+
+<Canvas>
+  <Story name="colors" story={stories.Colors}/>
+</Canvas>
+
+
+### Label color
+
+Please find an example of a spinner below when the `labelColor` prop is set to `"light"`. This will ensure proper contrast is 
+achieved on the label, when used in a dark background.
+
+<Canvas>
+  <Story name="label color" story={stories.LabelColor}/>
+</Canvas>
+
+
+### Motion
+
+When the `motion` prop is set to `false`, the motion will become static, and no animation will trigger.
+
+<Canvas>
+  <Story name="motion" story={stories.Motion}/>
+</Canvas>
+
+
+### Tracked
+
+When the `tracked` prop is set to `true`, the spinner will become tracked, for specific use cases where wait/loading times
+are predictable. The `tracked` prop cannot be true when the `gradient-grey` or `gradient-white` colors are set.
```

Comment:
Yeah, there aren't solid designs using a typical tracked animation type + @harpalsingh raised concerns about any sort of spinning gradient colour - as this may not be performative and technically feasible without the use of third party plug ins